### PR TITLE
Maintenance: Remove deprecation of `manager-api`'s `types` export

### LINKED ISSUE #65: [Bug]: Deprecated APIs and types are being used
### Describe the bug

The APIs used to register the manager-specific parts of the addon is deprecated and using some properties that TypeScript says are wrong. I don't have the full picture of how wrong it actually is, or what the non-deprecated APIs to migrate to would be, but we should probably change it.

This is also [in the documentation](https://storybook.js.org/docs/addons/writing-addons), so what ever we figure out we should remember to update the docs as well.

### Screenshots and/or logs

These are screenshots from just opening `manager.ts` from this repo. https://github.com/storybookjs/addon-kit/blob/main/src/manager.ts

image

### deprecated `types` import

image

### Tool definition

image

### Panel definition

image

### Tab definition

image

### PR DESCRIPTION
**Pull request by ndelangen**
Closes https://github.com/storybookjs/addon-kit/issues/65

## What I did

I removed the JSdoc deprecation tag from the comment above the `types` export from `manager-api`.

It should now no longer be marked as deprecated inside of addons/user's`manager.ts` files.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

Use the following API in an addon or `manager.ts` file, and check in an editor like vscode if the `types` export is marked as deprecated (it should NOT be, after this change)

```ts
import { types } from '@storybook/manager-api`;
```

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: code/lib/manager-api/src/index.tsx [hunk 1]
```diff
@@ -520,7 +520,6 @@ export { addons } from './lib/addons';
 /**
  * We need to rename this so it's not compiled to a straight re-export
  * Our globalization plugin can't handle an import and export of the same name in different lines
```
Comment (JReinhold) [inline code/lib/manager-api/src/index.tsx:522]: this comment still shows up to users on hover even though it should be internal, which is pretty unfortunate.

**Review by JReinhold — APPROVED** [review]

**Comment by JReinhold** [comment]
@ndelangen this doesn't close that issue completely, it just fixes the first issue of the deprecation notice. There are other problems mentioned in that issue that still needs to be addresseed.

File: code/lib/manager-api/src/index.tsx (hunk 1 above)
Comment (ndelangen) [reply code/lib/manager-api/src/index.tsx:522]: OK, I can change it to not be shown.

**Comment by ndelangen** [comment]
Thanks for calling that out, I can fix both issues in this PR

**Comment by ndelangen** [comment]
I tested this locally via setting `ui/.storybook/manager.tsx` to this content:
```tsx
import React, { memo, useCallback, useEffect } from 'react';
import { addons, types, useGlobals, useStorybookApi } from '@storybook/manager-api';
import startCase from 'lodash/startCase.js';
import { Icons, IconButton } from '@storybook/components';

addons.setConfig({
  sidebar: {
    renderLabel: ({ name, type }) => (type === 'story' ? name : startCase(name)),
  },
});

const PARAM_KEY = 'null';
const ADDON_ID = 'null';
const TOOL_ID = 'null';

export const Tool = memo(function MyAddonSelector() {
  const [globals, updateGlobals] = useGlobals();
  const api = useStorybookApi();

  const isActive = [true, 'true'].includes(globals[PARAM_KEY]);

  const toggleMyTool = useCallback(() => {
    updateGlobals({
      [PARAM_KEY]: !isActive,
    });
  }, [isActive]);

  useEffect(() => {
    api.setAddonShortcut(ADDON_ID, {
      label: 'Toggle Addon [8]',
      defaultShortcut: ['8'],
      actionName: 'myaddon',
      showInMenu: false,
      action: toggleMyTool,
    });
  }, [toggleMyTool, api]);

  return (
    <IconButton
      key={TOOL_ID}
      active={isActive}
      title="Apply outlines to the preview"
      onClick={toggleMyTool}
    >
      <Icons icon="lightning" />
    </IconButton>
  );
});

addons.register(ADDON_ID, () => {
  // Register the tool
  addons.add(TOOL_ID, {
    type: types.TOOL,
    title: 'My addon',
    match: ({ viewMode }) => !!(viewMode && viewMode.match(/^(story|docs)$/)),
    render: Tool,
  });
  addons.add(TOOL_ID, {
    type: types.TOOL,
    title: 'My addon',
    match: ({ viewMode }) => !!(viewMode && viewMode.match(/^(story|docs)$/)),
    render: ({ active }) => undefined,
  });
});
```

**Pull request merged**